packages/zoltan2/core/src/input/Zoltan2_Adapter.hpp

```diff
@@ -243,11 +243,31 @@ template <typename User>
   // This function does not write edge info to the graph file, as the
   // BaseAdapter does not know about edge info; it writes
   // only the Chaco header and vertex weights (if applicable).
-  void generateWeightFileOnly(const char* fileprefix, 
+  void generateWeightFileOnly(const char* fileprefix,
                               const Teuchos::Comm<int> &comm) const;
 
 };
 
+template <typename User>
+class AdapterWithCoords : public BaseAdapter<User>
+{
+public:
+  virtual void getCoordinatesKokkosView(
```

Comment:
It probably would make sense to put all getCoordinates* methods into this class (rather than only the Kokkos-based methods)?